platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/status/SyncPlatformStatusStateMachine.java

```diff
@@ -0,0 +1,203 @@
+                return currentStatusLogic.processCatastrophicFailureAction((CatastrophicFailureAction) action);
+            } else if (actionClass == DoneReplayingEventsAction.class) {
+                return currentStatusLogic.processDoneReplayingEventsAction((DoneReplayingEventsAction) action);
+            } else if (actionClass == FallenBehindAction.class) {
+                return currentStatusLogic.processFallenBehindAction((FallenBehindAction) action);
+            } else if (actionClass == FreezePeriodEnteredAction.class) {
+                return currentStatusLogic.processFreezePeriodEnteredAction((FreezePeriodEnteredAction) action);
+            } else if (actionClass == ReconnectCompleteAction.class) {
+                return currentStatusLogic.processReconnectCompleteAction((ReconnectCompleteAction) action);
+            } else if (actionClass == SelfEventReachedConsensusAction.class) {
+                return currentStatusLogic.processSelfEventReachedConsensusAction(
+                        (SelfEventReachedConsensusAction) action);
+            } else if (actionClass == StartedReplayingEventsAction.class) {
+                return currentStatusLogic.processStartedReplayingEventsAction((StartedReplayingEventsAction) action);
+            } else if (actionClass == StateWrittenToDiskAction.class) {
+                return currentStatusLogic.processStateWrittenToDiskAction((StateWrittenToDiskAction) action);
+            } else if (actionClass == TimeElapsedAction.class) {
+                return currentStatusLogic.processTimeElapsedAction((TimeElapsedAction) action);
+            } else {
+                throw new IllegalArgumentException(
+                        "Unknown action type: " + action.getClass().getName());
+            }
+        } catch (final IllegalPlatformStatusException e) {
+            logger.error(EXCEPTION.getMarker(), e.getMessage(), e);
+            return null;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public void processStatusAction(@NonNull final PlatformStatusAction action) {
+        Objects.requireNonNull(action);
+
+        final PlatformStatusLogic newLogic = getNewLogic(action);
+
+        if (newLogic == null || newLogic == currentStatusLogic) {
+            // if status didn't change, there isn't anything to do
+            return;
+        }
+
+        final String previousStatusName = currentStatusLogic.getStatus().name();
+        final String newStatusName = newLogic.getStatus().name();
+
+        final String statusChangeMessage = "Platform spent %s time in %s. Now in %s"
+                .formatted(Duration.between(currentStatusStartTime, time.now()), previousStatusName, newStatusName);
+
+        logger.info(
+                PLATFORM_STATUS.getMarker(),
+                () -> new PlatformStatusPayload(statusChangeMessage, previousStatusName, newStatusName).toString());
+
+        notificationEngine.dispatch(
```

Comment:
I would move this to the very last thing done in this method. It is synchronous, it could delay updating the actual status and the start time.

Reply:
Done